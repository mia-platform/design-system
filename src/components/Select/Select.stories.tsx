/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Meta, StoryObj } from '@storybook/react'
import { FaDiamond } from 'react-icons/fa6'
import { Flex } from 'antd'

import { Icon } from '../Icon'
import { Select } from '.'
import { Typography } from '../Typography'

const meta = {
  component: Select,
  args: {
    placeholder: 'Placeholder...',
  },
  argTypes: {
  },
} satisfies Meta<typeof Select>

export default meta
type Story = StoryObj<typeof meta>

const options = [
  ...Array(5).keys(),
].map((id) => ({
  value: `value ${id + 1}`,
}))

export const Default: Story = {
  args: {
    options,
  },
}

export const Borderless: Story = {
  args: {
    options,
    appearance: Select.Appearance.Borderless,
  },
}

export const ReadOnly: Story = {
  args: {
    options,
    isReadOnly: true,
  },
}

export const Error: Story = {
  args: {
    options,
    isError: true,
  },
}

export const Disabled: Story = {
  args: {
    options,
    isDisabled: true,
  },
}

export const AllowClear: Story = {
  args: {
    options,
    defaultValue: options[0].value,
    allowClear: true,
  },
}

export const CustomLabels: Story = {
  args: {
    options: options.map(({ value }) => ({
      value,
      label: (
        <Flex align="center" gap={8}>
          <Icon component={FaDiamond} size={16} />
          <Typography.BodyM isBold>{value}</Typography.BodyM>
        </Flex>
      ),
    })),
  },
}

export const Multiple: Story = {
  args: {
    options,
    multiple: true,
  },
}

export const MultipleDefaultValue: Story = {
  args: {
    defaultValue: [options[0].value, options[1].value],
    options,
    multiple: true,
  },
}

export const MultipleDisabled: Story = {
  args: {
    defaultValue: [options[0].value, options[1].value],
    options,
    multiple: true,
    isDisabled: true,
  },
}
