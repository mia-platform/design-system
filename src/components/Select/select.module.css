
.select {
  & :global(.mia-platform-select-arrow) {
    color:  var(--palette-text-neutral-main, #636363);;
  }

  & :global(.mia-platform-select-clear) {
    opacity: 1 !important;
    margin-inline-end: 20px;
  }

  & :global(.mia-platform-tag) {
    margin-inline-end: 4px;
    border-radius:  var(--shape-border-radius-sm, 2px);
  }

  &:not(.disabled) :global(.mia-platform-tag) {
    color: var(--palette-text-neutral-main, #636363);
    background-color: var(--palette-background-neutral-300, #F9F9F9);
    border-color: var(--palette-action-secondary-bold, #CDCDCD);
  }

}

.readOnly {
  & :global(.mia-platform-select-selector) {
    background: transparent !important;
    border: none !important;
  }
}

.multiple {
  & :global(.mia-platform-select-selector) {
    padding-inline: var(--spacing-padding-xs, 4px);
  }
}

.disabled {
  & :global(.mia-platform-select-selector) {
    border-color: var(--palette-action-disabled-bold, #E5E5E5) !important;
  }

  & :global(.mia-platform-select-arrow) {
    color: var(--palette-text-neutral-subtler, #636363);
  }
  & :global(.mia-platform-tag) {
    color: var(--palette-text-neutral-subtler, #ACACAC);
    background-color: var(--palette-background-neutral-200, #F2F2F2);
    border-color: var(--palette-action-disabled-bold, #E5E5E5);
  }
}
