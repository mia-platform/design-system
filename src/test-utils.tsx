/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { RenderOptions, RenderResult, render } from '@testing-library/react'
import { ConfigProvider } from 'antd'
import { ReactNode } from 'react'

import { ThemeContext, defaultTheme } from './components/ThemeProvider'
import { ANT_PREFIX } from './components/ThemeProvider/Ant'

/**
 * Wraps a React node with a series of providers for testing purposes.
 *
 * @remarks It adds the default theme to components
 * @remarks It adds the default Ant class prefix
 * @remarks It removes the css-do-not-override classes from Ant Design components
 *
 * @param {ReactNode} children - The React node to be wrapped.
 * @returns {ReactNode} The wrapped React node.
 */
const TestWrapper = ({ children }: {children: ReactNode}): ReactNode => (
  <ThemeContext.Provider value={defaultTheme}>
    <ConfigProvider prefixCls={ANT_PREFIX} theme={{ hashed: false }}>
      {children}
    </ConfigProvider>
  </ThemeContext.Provider>
)

/**
 * Custom render function for testing React components.
 *
 * @param {ReactNode} element - The React node to be rendered.
 * @param {RenderOptions} [options] - Options for rendering.
 * @returns {RenderResult} The result of rendering the React node.
 */
const customRender = (
  element: ReactNode,
  options?: Omit<RenderOptions, 'wrapper'>,
): RenderResult => render(element, { wrapper: TestWrapper, ...options })

export * from '@testing-library/react'
export {
  customRender as render,
  TestWrapper,
}
