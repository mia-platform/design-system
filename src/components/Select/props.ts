/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, ReactNode } from 'react'
import type { SelectProps as AntdSelectProps } from 'antd'
import { FilterFunc } from 'rc-select/lib/Select'
import { FlattenOptionData } from 'rc-select/lib/interface'

import { BaseInputProps } from '../BaseInput/props'
import { Mode } from './Select.types'
import { SearchSelectHandler } from '../Search/props'

export type SelectItem<ValueType> = {
  value: ValueType
  label?: ReactNode
  [name: string]: unknown;
}

export type SelectChangeHandler<ValueType> = (
  value: ValueType,
  option: SelectItem<ValueType> | SelectItem<ValueType>[]
) => void

export type SelectSelectHandler<ValueType> = (
    value: ValueType extends (infer E)[] ? E : ValueType,
    option: SelectItem<ValueType>
  ) => void

export type OptionRender<ValueType> = (
  option: FlattenOptionData<SelectItem<ValueType>>,
  info: {index: number}
) => ReactNode

export type FilterOption<ValueType> = boolean | FilterFunc<SelectItem<ValueType>>

export type SelectProps<ValueType = unknown> = BaseInputProps & {

  /**
   * The id of the input.
   */
  id?: string

  /**
   * The input content value.
   */
  value?: ValueType

  /**
   * The default value for the input.
   */
  defaultValue?: ValueType

  /**
  * Select options.
   */
  options: SelectItem<ValueType>[]

  /**
   * If allow to remove input content with clear icon.
   */
  allowClear?: boolean

  /**
   * If allow to remove input content with clear icon.
   */
  maxTagCount?: number | 'responsive',

  /**
   * Placeholder for not showing tags when maxTagCount is passed
   */
  maxTagPlaceholder?: AntdSelectProps['maxTagPlaceholder'],

  /**
   * Callback when user input.
   */
  onChange?: SelectChangeHandler<ValueType>

  /**
   * Callback when user input.
   */
  onClear?: () => void

  /**
   * Callback when an option is selected.
   */
  onSelect?: SearchSelectHandler<ValueType>

  /**
   * Callback when an option is selected.
   */
  onDeselect?: SelectSelectHandler<ValueType>

  /**
   * Callback function that is fired when input changed.
   */
  onSearch?: (value: string) => void

  /**
   * @deprecated use the `mode` property to use the Select with multiple selection
   * If multiple values are allowed.
   */
  isMultiple?: boolean

  /**
   * If tags mode is enabled.
   */
  mode?: Mode

  /**
   * Separator used to tokenize (es: ","), only applies when mode="tags"
   */
  tokenSeparators?: string[]

  /**
   * Customize the rendering dropdown options.
   */
  optionRender?: OptionRender<ValueType>

  /**
   * If true, filter options by input, if function, filter options against it.
   * The function will receive two arguments, inputValue and option, if the function
   * returns true, the option will be included in the filtered set; otherwise, it will be excluded.
   */
  filterOption?: FilterOption<ValueType>

  /**
   * The prop name to filter options. Default is 'value'.
   */
  optionFilterProp?: string

  /**
   * Customize the rendering of the dropdown.
   */
  dropdownRender?: (menu: ReactNode) => ReactElement

  /**
   * Determine whether the popup menu and the select input should have the same width.
   * Default set `min-width` of the popup to match the input.
   * This is ignored when the value is less than the select input's width.
   * Setting this to `false` will disable the virtual scroll
   */
  popupMatchSelectWidth?: boolean | number

  /**
   * Parent Node which the selector should be rendered to. Default to body.
  */
  getPopupContainer?: (triggerNode: HTMLElement) => HTMLElement

  /**
   * The position where the selection box pops up
   */
  placement?: 'bottomLeft' | 'bottomRight' | 'topLeft' | 'topRight'

  /**
   * Classes for the select.
   */
  className?: string
}
