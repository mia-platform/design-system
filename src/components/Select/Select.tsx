/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Select as AntSelect, SelectProps as AntSelectProps } from 'antd'
import { PiCaretDown, PiCaretUp, PiCheck } from 'react-icons/pi'
import { ReactElement, useMemo, useState } from 'react'
import classnames from 'classnames'

import { BaseInput, defaults as baseInputDefaults } from '../BaseInput/BaseInput'
import { SelectItem, SelectProps } from './props'
import { Appearance } from '../BaseInput/types'
import { Icon } from '../Icon'
import { Tag } from '../Tag/Tag'
import styles from './select.module.css'

/**
 * A UI element to insert long text content in a form.
 *
 * @link https://ant.design/components/Autocomplete
 * @returns {Select} Select component
 */

const DEFAULT_ICON_SIZE = 16 as never

export const defaults = {
  ...baseInputDefaults,
}

const menuItemSelectedIcon = <Icon component={PiCheck} size={16} />

const tagRender = ({ value, onClose, closable }: {
  label: React.ReactNode;
  value: string;
  closable: boolean;
  onClose: (event?: React.MouseEvent<HTMLElement, MouseEvent>) => void;
}): ReactElement => {
  return (
    <Tag closeIcon={closable} isBordered onClose={onClose}>
      {value}
    </Tag>
  )
}

export const Select = <ValueType, >(
  {
    appearance = defaults.type,
    value,
    defaultValue,
    isDisabled,
    isReadOnly,
    isFullWidth = defaults.isFullWidth,
    isError,
    placeholder,
    allowClear,
    options,
    inputRef,
    onChange,
    onClear,
    onSelect,
    onDeselect,
    isMultiple,
  }: SelectProps<ValueType>) : ReactElement => {
  const [open, setOpen] = useState(false)

  const className = useMemo(() => classnames([
    styles.select,
    isMultiple && styles.multiple,
    isDisabled && styles.disabled,
    isReadOnly && styles.readOnly,
  ]), [isDisabled, isMultiple, isReadOnly])

  const suffixIcon = useMemo(() => !isReadOnly && (
    <Icon component={open ? PiCaretUp : PiCaretDown} size={DEFAULT_ICON_SIZE} />
  ), [isReadOnly, open])

  return (
    <BaseInput <AntSelectProps<ValueType, SelectItem<ValueType>>>
      allowClear={allowClear}
      appearance={appearance}
      className={className}
      component={AntSelect}
      defaultValue={defaultValue}
      inputRef={inputRef}
      isDisabled={isDisabled}
      isError={isError}
      isFullWidth={isFullWidth}
      isReadOnly={isReadOnly}
      menuItemSelectedIcon={menuItemSelectedIcon}
      mode={isMultiple ? 'multiple' : undefined}
      options={options}
      placeholder={placeholder}
      showSearch={false}
      suffixIcon={suffixIcon}
      tagRender={tagRender}
      value={value}
      onChange={onChange}
      onClear={onClear}
      onDeselect={onDeselect}
      onDropdownVisibleChange={setOpen}
      onSelect={onSelect}
    />
  )
}

Select.Appearance = Appearance
