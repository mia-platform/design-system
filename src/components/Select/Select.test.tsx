/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'
import { within } from '@testing-library/react'

import { render, screen, userEvent } from '../../test-utils'
import { Select } from './Select'

const options = [
  ...Array(5).keys(),
].map((id) => ({
  value: `value ${id + 1}`,
}))

describe('Select Component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders correctly', () => {
    const { asFragment } = render(<Select options={options} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Borderless renders correctly', () => {
    const { asFragment } = render(<Select appearance={Select.Appearance.Borderless} options={options} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Read only renders correctly', () => {
    const { asFragment } = render(<Select isReadOnly options={options} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Error renders correctly', () => {
    const { asFragment } = render(<Select isError options={options} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Disabled renders correctly', () => {
    const { asFragment } = render(<Select isDisabled options={options} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Should call event handler on Change', async() => {
    const onChange = jest.fn()

    render(<Select options={options} onChange={onChange} />)

    const input = screen.getByRole<HTMLInputElement>('combobox')

    await userEvent.click(input)

    const option = screen.getByTitle(options[0].value)
    await userEvent.click(option)

    expect(onChange).toHaveBeenCalledWith(options[0].value, options[0])
  })

  test('Allow clear should clear input', async() => {
    const onClear = jest.fn()

    render(
      <Select
        allowClear
        defaultValue={options[0].value}
        options={options}
        onClear={onClear}
      />
    )

    const clearIcon = screen.getByRole('img', { name: 'close-circle', hidden: true })

    await userEvent.click(clearIcon)

    expect(onClear).toHaveBeenCalled()
  })

  test('Should call select handlers onChange and onSelect if multiple', async() => {
    const onSelect = jest.fn()
    const onChange = jest.fn()

    render(
      <Select
        isMultiple
        options={options}
        onChange={onChange}
        onSelect={onSelect}
      />
    )

    const input = screen.getByRole<HTMLInputElement>('combobox')

    await userEvent.click(input)

    const option1 = screen.getByTitle(options[0].value)
    const option2 = screen.getByTitle(options[1].value)
    await userEvent.click(option1)

    expect(onSelect).toHaveBeenCalledWith(options[0].value, options[0])
    expect(onChange).toHaveBeenCalledWith(
      [options[0].value],
      [options[0]]
    )
    await userEvent.click(option2)

    expect(onSelect).toHaveBeenCalledWith(options[1].value, options[1])
    expect(onChange).toHaveBeenCalledWith(
      [options[0].value, options[1].value],
      [options[0], options[1]]
    )
  })

  test('Should call select handlers onChange and onDeselect on tag click if multiple', async() => {
    const onDeselect = jest.fn()
    const onChange = jest.fn()

    render(
      <Select
        defaultValue={[options[0].value]}
        isMultiple
        options={options}
        onChange={onChange}
        onDeselect={onDeselect}
      />
    )

    const option = screen.getByText(options[0].value)
    const closeIcon = within(option).getByRole('img', { name: 'close' })
    await userEvent.click(closeIcon)

    expect(onDeselect).toHaveBeenCalledWith(options[0].value, options[0])
    expect(onChange).toHaveBeenCalledWith([], [])
  })

  test('Should show tag placeholder if in multiple mode', async() => {
    const onDeselect = jest.fn()
    const onChange = jest.fn()

    render(
      <Select
        defaultValue={[options[0].value, options[1].value, options[2].value]}
        isMultiple
        maxTagCount={1}
        maxTagPlaceholder={`number of collapsed tag: ${options.length - 1}`}
        options={options}
        onChange={onChange}
        onDeselect={onDeselect}
      />
    )

    const placeholder = screen.getByText(`number of collapsed tag: ${options.length - 1}`)
    expect(placeholder).toBeVisible()
  })

  test('Should use option render opening the select dropdown', async() => {
    const optionsWithDescription = [
      ...Array(5).keys(),
    ].map((id) => ({
      value: `Custom value ${id + 1}`,
      id: `value ${id + 1}`,
      description: `description ${id + 1}`,
    }))
    render(
      <Select
        defaultValue={optionsWithDescription[0].id}
        optionRender={(option) => (
          <span>
            {option.data.value}
            {' - '}
            {option.data.description as ReactNode}
          </span>
        )}
        options={optionsWithDescription}
      />
    )

    const defaultSelection = screen.getByTitle('value 1')
    expect(defaultSelection).toBeInTheDocument()

    const input = screen.getByRole<HTMLInputElement>('combobox')
    await userEvent.click(input)

    expect(screen.getByTitle('Custom value 1')).toBeInTheDocument()
    expect(screen.getByText(/description 1/i)).toBeInTheDocument()
    expect(screen.getByTitle('Custom value 2')).toBeInTheDocument()
    expect(screen.getByText(/description 2/i)).toBeInTheDocument()
    expect(screen.getByTitle('Custom value 3')).toBeInTheDocument()
    expect(screen.getByText(/description 3/i)).toBeInTheDocument()
    expect(screen.getByTitle('Custom value 4')).toBeInTheDocument()
    expect(screen.getByText(/description 4/i)).toBeInTheDocument()
    expect(screen.getByTitle('Custom value 5')).toBeInTheDocument()
    expect(screen.getByText(/description 5/i)).toBeInTheDocument()
  })

  test('Should call onSearch when typing', async() => {
    const onSearch = jest.fn()
    render(
      <Select
        options={options}
        onSearch={onSearch}
      />
    )
    const input = screen.getByRole<HTMLInputElement>('combobox')
    await userEvent.click(input)

    await userEvent.type(input, 'test')
    expect(input).toHaveValue('test')

    expect(onSearch).toHaveBeenCalledTimes(4)
    expect(onSearch).toHaveBeenNthCalledWith(1, 't')
    expect(onSearch).toHaveBeenNthCalledWith(2, 'te')
    expect(onSearch).toHaveBeenNthCalledWith(3, 'tes')
    expect(onSearch).toHaveBeenNthCalledWith(4, 'test')
  })

  test('Should filter option when filterOption set to true', async() => {
    render(
      <Select
        filterOption
        options={options}
      />
    )
    const input = screen.getByRole<HTMLInputElement>('combobox')
    await userEvent.click(input)

    await userEvent.type(input, '5')
    expect(input).toHaveValue('5')

    expect(screen.getByTitle('value 5')).toBeInTheDocument()
    expect(screen.queryByTitle('value 1')).not.toBeInTheDocument()
  })

  test('Should filter option when filterOption set to a function', async() => {
    const filterOption = jest.fn((input, option) => {
      return option.value.toLowerCase().includes(input.toLowerCase())
    })
    render(
      <Select
        filterOption={filterOption}
        options={options}
      />
    )
    const input = screen.getByRole<HTMLInputElement>('combobox')
    await userEvent.click(input)

    await userEvent.type(input, '5')
    expect(input).toHaveValue('5')

    expect(filterOption).toHaveBeenCalledTimes(options.length)
    expect(filterOption).toHaveBeenNthCalledWith(1, '5', options[0])
    expect(filterOption).toHaveBeenNthCalledWith(2, '5', options[1])
    expect(filterOption).toHaveBeenNthCalledWith(3, '5', options[2])
    expect(filterOption).toHaveBeenNthCalledWith(4, '5', options[3])
    expect(filterOption).toHaveBeenNthCalledWith(5, '5', options[4])
  })
})
