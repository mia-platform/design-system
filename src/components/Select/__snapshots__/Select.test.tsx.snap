// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component Borderless renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mia-platform-select mia-platform-select-borderless input fullWidth select mia-platform-select-single mia-platform-select-show-arrow"
  >
    <div
      class="mia-platform-select-selector"
    >
      <span
        class="mia-platform-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="mia-platform-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          readonly=""
          role="combobox"
          style="opacity: 0;"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="mia-platform-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="mia-platform-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Select Component Disabled renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mia-platform-select mia-platform-select-outlined input fullWidth disabled select disabled mia-platform-select-single mia-platform-select-show-arrow mia-platform-select-disabled"
  >
    <div
      class="mia-platform-select-selector"
    >
      <span
        class="mia-platform-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="mia-platform-select-selection-search-input"
          disabled=""
          id="rc_select_TEST_OR_SSR"
          readonly=""
          role="combobox"
          style="opacity: 0;"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="mia-platform-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="mia-platform-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Select Component Error renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mia-platform-select mia-platform-select-outlined mia-platform-select-status-error input fullWidth select mia-platform-select-single mia-platform-select-show-arrow"
  >
    <div
      class="mia-platform-select-selector"
    >
      <span
        class="mia-platform-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="mia-platform-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          readonly=""
          role="combobox"
          style="opacity: 0;"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="mia-platform-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="mia-platform-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Select Component Read only renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mia-platform-select mia-platform-select-outlined input fullWidth readonly select readOnly mia-platform-select-single mia-platform-select-show-arrow"
  >
    <div
      class="mia-platform-select-selector"
    >
      <span
        class="mia-platform-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="mia-platform-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          readonly=""
          role="combobox"
          style="opacity: 0;"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="mia-platform-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="mia-platform-select-arrow"
      style="user-select: none;"
      unselectable="on"
    />
  </div>
</DocumentFragment>
`;

exports[`Select Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mia-platform-select mia-platform-select-outlined input fullWidth select mia-platform-select-single mia-platform-select-show-arrow"
  >
    <div
      class="mia-platform-select-selector"
    >
      <span
        class="mia-platform-select-selection-search"
      >
        <input
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="mia-platform-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          readonly=""
          role="combobox"
          style="opacity: 0;"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="mia-platform-select-selection-placeholder"
      />
    </div>
    <span
      aria-hidden="true"
      class="mia-platform-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="16"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
